fq.go

```diff
@@ -0,0 +1,80 @@
+package fq
+
+import (
+	"math"
+	"sync"
+)
+
+type fqscheduler struct {
+	lock   *sync.Mutex
```

Comment:
Why use a pointer here?  Directly embedding the mutex works just fine.

Reply:
good point, changed